Traceback (most recent call last):
  File "/home/runner/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/runner/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/runner/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import anndata
import numpy as np
import tempfile
from cellarr import build_cellarrdataset, CellArrDataset, MatrixOptions

# Create a temporary directory
tempdir = tempfile.mkdtemp()

# # Read AnnData objects
# adata1 = anndata.read_h5ad("path/to/object1.h5ad", "r")
# # or just provide the path
# adata2 = "path/to/object2.h5ad"

# Build CellArrDataset
dataset = build_cellarrdataset(
    output_path=tempdir,
    files=[adata1, adata2],
    matrix_options=MatrixOptions(dtype=np.float32),
)

print(dataset)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mTypeError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[2], line 15[0m
[1;32m      7[0m tempdir [38;5;241m=[39m tempfile[38;5;241m.[39mmkdtemp()
[1;32m      9[0m [38;5;66;03m# # Read AnnData objects[39;00m
[1;32m     10[0m [38;5;66;03m# adata1 = anndata.read_h5ad("path/to/object1.h5ad", "r")[39;00m
[1;32m     11[0m [38;5;66;03m# # or just provide the path[39;00m
[1;32m     12[0m [38;5;66;03m# adata2 = "path/to/object2.h5ad"[39;00m
[1;32m     13[0m 
[1;32m     14[0m [38;5;66;03m# Build CellArrDataset[39;00m
[0;32m---> 15[0m dataset [38;5;241m=[39m [43mbuild_cellarrdataset[49m[43m([49m
[1;32m     16[0m [43m    [49m[43moutput_path[49m[38;5;241;43m=[39;49m[43mtempdir[49m[43m,[49m
[1;32m     17[0m [43m    [49m[43mfiles[49m[38;5;241;43m=[39;49m[43m[[49m[43madata1[49m[43m,[49m[43m [49m[43madata2[49m[43m][49m[43m,[49m
[1;32m     18[0m [43m    [49m[43mmatrix_options[49m[38;5;241;43m=[39;49m[43mMatrixOptions[49m[43m([49m[43mdtype[49m[38;5;241;43m=[39;49m[43mnp[49m[38;5;241;43m.[39;49m[43mfloat32[49m[43m)[49m[43m,[49m
[1;32m     19[0m [43m)[49m
[1;32m     21[0m [38;5;28mprint[39m(dataset)

File [0;32m~/work/cellarr/cellarr/.tox/docs/lib/python3.9/site-packages/cellarr/build_cellarrdataset.py:222[0m, in [0;36mbuild_cellarrdataset[0;34m(files, output_path, gene_annotation, sample_metadata, cell_metadata, sample_metadata_options, cell_metadata_options, gene_annotation_options, matrix_options, optimize_tiledb, num_threads)[0m
[1;32m    218[0m [38;5;66;03m####[39;00m
[1;32m    219[0m [38;5;66;03m## Writing gene annotation file[39;00m
[1;32m    220[0m [38;5;66;03m####[39;00m
[1;32m    221[0m [38;5;28;01mif[39;00m gene_annotation [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[0;32m--> 222[0m     [43mwarnings[49m[38;5;241;43m.[39;49m[43mwarn[49m[43m([49m
[1;32m    223[0m [43m        [49m[43m([49m
[1;32m    224[0m [43m            [49m[38;5;124;43m"[39;49m[38;5;124;43mScanning all files for feature ids (e.g. gene symbols) and cell annotations, this may take long, [39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m    225[0m [43m            [49m[38;5;124;43m"[39;49m[38;5;124;43mPlease also make sure you have enough memory.[39;49m[38;5;124;43m"[39;49m[43m,[49m
[1;32m    226[0m [43m        [49m[43m)[49m[43m,[49m
[1;32m    227[0m [43m        [49m[38;5;167;43;01mUserWarning[39;49;00m[43m,[49m
[1;32m    228[0m [43m    [49m[43m)[49m
[1;32m    230[0m     gene_set [38;5;241m=[39m uad[38;5;241m.[39mscan_for_features(files_cache)
[1;32m    231[0m     gene_set [38;5;241m=[39m [38;5;28msorted[39m(gene_set)

[0;31mTypeError[0m: expected string or bytes-like object

